src/axolotl/utils/trainer.py

```diff
@@ -169,116 +170,181 @@ def add_length(sample):
+        LOG.debug(f"max_input_len: {max_len}", main_process_only=True)
     drop_long = partial(
         drop_long_seq,
+        sequence_len=sequence_len,
+        min_sequence_len=min_sample_len or 2,
+    )
+    len_pre_drop = len(dataset)
+    dataset = dataset.filter(
+        drop_long,
+        num_proc=dataset_processes,
+        load_from_cache_file=load_from_cache_file,
+        desc=f"Dropping Long Sequences From {ds_split_name} Dataset",
+    )
+    dropped_rows = len_pre_drop - len(dataset)
+    if dropped_rows > 0:
+        LOG.warning(f"Dropped {dropped_rows} rows from {ds_split_name} dataset")
+        if not ignore_long_sequences:
+            error_message = (
+                f"Found {dropped_rows} sequences longer than {sequence_len} tokens in {ds_split_name} Dataset. "
+                f"Please either increase --sequence_len or set --drop_long_sequences to True to drop and ignore such sequences."
+            )
+            if max_len is not None:
+                error_message += f" Longest sequence is {max_len} tokens."
+            raise ValueError(error_message)
+
+    if drop_no_outputs:
+        len_pre_drop = len(dataset)
+        dataset = dataset.filter(
+            drop_samples_with_no_outputs,
+            num_proc=dataset_processes,
+            load_from_cache_file=load_from_cache_file,
+            desc="Dropping Sequences Without Outputs",
+        )
+        dropped_rows = len_pre_drop - len(dataset)
+        if dropped_rows > 0:
+            LOG.warning(
+                f"Dropped {dropped_rows} rows with no outputs from {ds_split_name} Dataset"
+            )
+    return dataset
+
+
+def process_datasets_for_packing(cfg, train_dataset, eval_dataset):
+    if (
+        cfg.is_mistral_derived_model and cfg.flash_attention
+    ) or cfg.model_config_type == "mamba":
+        LOG.info("dropping attention_mask column")
+        train_dataset = train_dataset.remove_columns("attention_mask")
+        if eval_dataset:
+            eval_dataset = eval_dataset.remove_columns("attention_mask")
+
+    if cfg.model_config_type == "falcon":
+        LOG.info("dropping token_type_ids column if it exists")
+        if "token_type_ids" in train_dataset.column_names:
+            train_dataset = train_dataset.remove_columns("token_type_ids")
+        if eval_dataset and "token_type_ids" in eval_dataset.column_names:
+            eval_dataset = eval_dataset.remove_columns("token_type_ids")
+
+    train_dataset = _maybe_drop_sequences(
+        dataset=train_dataset,
+        ds_split_name="Train",
         sequence_len=cfg.sequence_len,
-        min_sequence_len=cfg.min_sample_len or 2,
+        min_sample_len=cfg.min_sample_len or 2,
+        ignore_long_sequences=cfg.drop_long_sequences,
+        drop_no_outputs=True,
+        dataset_processes=cfg.dataset_processes,
+        load_from_cache_file=not cfg.is_preprocess,
+        show_lengths_summary=True,
```

Comment:
```suggestion
        show_lengths_summary=cfg.is_preprocess,
```
the outputs are likely going to be really wonky if run on multi-process/gpu. since preprocessing is typically single process, it's going to be safer.

Reply:
I generally agree, I'll make this change for now, but considering this function is called under with zero first, there is no wonkiness